src/server/ideController.py

```diff
@@ -5,25 +6,54 @@
 from genshi.template import TemplateLoader
 import uuid  # XXX Temp for fake session id... Could be used for real?
 
+def is_valid_path(path):
+  normalized = os.path.normpath(path)
+  # .. is illegal
+  # / is illegal
+  return (normalized == path) and (not path.endswith('/')) 
+
+def is_valid_changes(changes):
+  # Check for 'count' or 'content' in keys
+  return all(
+          (('count' in c == 'content' in c) and
+           'pos' in c and
+           'type' in c and
+           c['type'] in (1, -1))
+          for c in changes)
+
+def create_argument_error_msg(arg):
+  return {code: 400, message: "Invalid argument provided : " + arg}
+
+
+def create_file_version_dict(filename, version, content):
+  return {'file':    filename,
+          'vers':    version,
+          'content': content}
+
+def create_tree_nodes_dict(nodes):
+  return {'nodes': [{'node': name,
+                     'isDir': is_dir}
+                     for (name, is_dir) in nodes]}
 
 class IDEController(object):
   """
   Controller of the IDE/Editing part
   """
 
-  def __init__(self, template_path, logger):
+  IDE_HTML_TEMPLATE = 'edit.html'
+
+  def __init__(self, app, template_path, logger):
     """
     IDEController initialiser
 
+    @param app: The core application
```

Comment:
Add the `@type cide.app.python.core` type in docstring if possible. That's nice for people using autocompletion (like me!)